custom/icds_reports/utils.py

```diff
@@ -1003,27 +998,89 @@ def get_prevalence_of_undernutrition_data_chart(config, loc_level):
     }
 
 
-@quickcache(['config', 'loc_level'], timeout=24 * 60 * 60)
 def get_prevalence_of_undernutrition_sector_data(config, loc_level):
+    group_by = ['%s_name' % loc_level]
+    if loc_level == LocationTypes.SUPERVISOR:
+        config['aggregation_level'] += 1
+        group_by.append('%s_name' % LocationTypes.AWC)
+
+    config['month'] = datetime(*config['month'])
+    data = AggChildHealthMonthly.objects.filter(
+        **config
+    ).values(
+        *group_by
+    ).annotate(
+        moderately_underweight=Sum('nutrition_status_moderately_underweight'),
+        severely_underweight=Sum('nutrition_status_severely_underweight'),
+        valid=Sum('valid_in_month'),
+    ).order_by('%s_name' % loc_level)
```

Comment:
This is going to be a huge query when aggregating by AWC. will probably timeout. is it going to be possible to paginate this query?